src/qtcore/qml/elements/Item.js

```diff
@@ -334,25 +333,10 @@ function QMLItem(meta) {
         this.css.height = newVal ? newVal + "px" : "auto";
     });
 
-    this.implicitHeight = 0;
-    this.implicitWidth = 0;
     this.spacing = 0;
-    this.x = 0;
-    this.y = 0;
-    this.anchors.margins = 0;
-    this.visible = true;
-    this.opacity = 1;
     this.$revertActions = [];
-    this.states = [];
-    this.transitions = [];
-    this.state = "";
-    this.transform = [];
-    this.rotation = 0;
-    this.scale = 1;
-    this.childrenRect.x = 0;
-    this.childrenRect.y = 0;
-    this.childrenRect.width = 0;
-    this.childrenRect.height = 0;
+    this.xChanged(0);
```

Comment:
Why do we want that? I guess we should call the function we need to be called directly, because the xChanged signal isn't supposed to be called from here, anyway.